examples/anonymize_dataset.py

```diff
@@ -0,0 +1,203 @@
+
+# %%
+# You can see that the subject ID was changed to a number (in this case, the
+# digit ``1```), and the recording dates have been shifted backward in time (as
+# indicated by the ``emptyroom`` session name). Anonymized subject IDs are
+# simply a permutation of the set :math:`\{1, 2, \ldots, N\}``, where :math:`N`
+# is the total number of participants (excluding the ``emptyroom``
+# pseudo-subject). Zero-padding will be added to the subject IDs as needed to
+# allow for convenient sorting.
+#
+# Limiting to specific data types
+# -------------------------------
+# By default, :func:`mne_bids.anonymize_dataset` will anonymize
+# electrophysiological data and anatomical MR scans (T1-weightes and FLASH).
+# You can limit which data types to convert using the ``datatypes`` keyword
+# argument. The parameter can be a string (e.g., ``'meg'``, ``'eeg'``,
+# ``'anat'``) or a list of such strings.
+
+shutil.rmtree(bids_root_anon)
+anonymize_dataset(
+    bids_root_in=bids_root,
+    bids_root_out=bids_root_anon,
+    datatypes='anat'  # Only anatomical data
+)
+print_dir_tree(bids_root_anon)
+
+# %%
+# Specifying time shift
+# ---------------------
+# Anonymization involves shifting the recording dates back in time. MNE-BIDS
+# will try to automatically choose a suitable time shift. You may also
+# explicitly specify by how many days you wish to shift the recording dates
+# back in time via the ``daysback`` parameter. To avoid the time shift, pass
+# ``daysback=0``.
+
+shutil.rmtree(bids_root_anon)
+anonymize_dataset(
+    bids_root_in=bids_root,
+    bids_root_out=bids_root_anon,
+    datatypes='meg',  # Only MEG data
+    daysback=10
+)
+print_dir_tree(bids_root_anon / 'sub-emptyroom')  # Easy to see effects here
+
+# %%
+# Specifying subject IDs
+# ----------------------
+# Anonymized subject IDs are automatically generated as unique 3-digit numbers.
+# You can control this behavior via the ``subject_mapping`` parameter. Set it
+# to ``None`` to avoid changing the subject IDs, e.g., in case they're already
+# anonymized. You can pass a dictionary that maps original subject IDs to the
+# anonymize IDs. Lastly, you can also pass a function that accepts a list of
+# original IDs and returns such a dictionary.
+
+shutil.rmtree(bids_root_anon)
+
+subject_mapping = {
+    'ABC123': 'anonymous',
+    'emptyroom': 'emptyroom'
+}
+
+anonymize_dataset(
+    bids_root_in=bids_root,
+    bids_root_out=bids_root_anon,
+    datatypes='meg',
+    subject_mapping=subject_mapping
+)
+print_dir_tree(bids_root_anon)
+
+# %%
+# Reproducibility
+# ---------------
+# Every time you run this function, the automatically-generated subject IDs and
+# the timeshift may differ (unless you excplicitly specify them as described
+# above), as they are determined randomly.
+#
+# To ensure results are reproducible across runs, you can pass the
+# ``random_state`` parameter, causing the random number generator to produce
+# the same results every time you execute the function.
```

Comment:
but then you lose anonymity when the script to anonymize is shared :)

Reply:
Only if you share it along with the original dataset … in which case you lose anonymity anyway, no?

Reply:
fair enough :)